code/modules/surgery/operations/_operation.dm

```diff
@@ -825,20 +852,37 @@ GLOBAL_DATUM_INIT(operations, /datum/operation_holder, new)
 			user = surgeon,
 			// Actual delay is capped - think of the excess time as being added to failure chance instead
 			delay = time * min(operation_args[OPERATION_SPEED], SURGERY_MODIFIER_FAILURE_THRESHOLD),
-			target = patient,
+			target = patient || operating_on,
 			extra_checks = CALLBACK(src, PROC_REF(operate_check), patient, operating_on, surgeon, tool, operation_args),
 			// You can only operate on one mob at a time without a hippocratic oath
-			interaction_key = HAS_TRAIT(surgeon, TRAIT_HIPPOCRATIC_OATH) ? patient : DOAFTER_SOURCE_SURGERY,
+			interaction_key = HAS_TRAIT(surgeon, TRAIT_HIPPOCRATIC_OATH) ? (patient || operating_on) : DOAFTER_SOURCE_SURGERY,
 		))
 			result |= ITEM_INTERACT_BLOCKING
-			update_surgery_mood(patient, SURGERY_STATE_FAILURE)
+			if (patient)
+				update_surgery_mood(patient, SURGERY_STATE_FAILURE)
 			break
 
-		if(ishuman(surgeon))
-			var/mob/living/carbon/human/surgeon_human = surgeon
-			surgeon_human.add_blood_DNA_to_items(patient.get_blood_dna_list(), ITEM_SLOT_GLOVES)
+		if (patient)
+			if (ishuman(surgeon))
+				var/mob/living/carbon/human/surgeon_human = surgeon
+				surgeon_human.add_blood_DNA_to_items(patient.get_blood_dna_list(), ITEM_SLOT_GLOVES)
+			else
+				surgeon.add_mob_blood(patient)
 		else
-			surgeon.add_mob_blood(patient)
+			var/list/dna_info = null
+			if (isbodypart(operating_on))
+				var/obj/item/bodypart/limb = operating_on
+				dna_info = limb.blood_dna_info?.Copy()
+			else if (isorgan(operating_on))
+				var/obj/item/organ/guts = operating_on
+				dna_info = guts.blood_dna_info?.Copy()
+
+			if (length(dna_info))
+				if (ishuman(surgeon))
+					var/mob/living/carbon/human/surgeon_human = surgeon
+					surgeon_human.add_blood_DNA_to_items(dna_info.Copy(), ITEM_SLOT_GLOVES)
+				else
+					surgeon.add_blood_DNA(dna_info.Copy())
```

Comment:
Can all this be separated out into its own proc, like `get_patient_dna(patient, operating_on)` or something

`surgeon.add_blood_DNA_to_items(get_patient_dna(patient, operating_on), ITEM_SLOT_GLOVES)`